fda/grid.py

```diff
@@ -677,92 +685,196 @@ def concatenate(self, other):
-                except AttributeError:
-                    pass
+            if ax[0].name == '3d':
+                for i in range(self.ndim_image):
+                    ax[i].set_xlabel(self.axes_labels[0])
+                    ax[i].set_ylabel(self.axes_labels[1])
+                    ax[i].set_zlabel(self.axes_labels[i + 2])
+            else:
+                for i in range(self.ndim_image):
+                    ax[i].set_xlabel(self.axes_labels[0])
+                    ax[i].set_ylabel(self.axes_labels[i + 1])
+
+    def _genereic_plotting_checks(self, fig=None, ax=None, nrows=None, ncols=None):
+        """Check the arguments passed to both :func:`plot <fda.grid.plot>` and :func:`scatter <fda.grid.scatter>`
+         methods of the FDatGrid object.
+
+        Args:
+            fig (figure object, optional): figure over with the graphs are plotted in case ax is not specified.
+                If None and ax is also None, the figure is initialized.
+            ax (list of axis objects, optional): axis over where the graphs are plotted. If None, see param fig.
+            nrows(int, optional): designates the number of rows of the figure to plot the different dimensions of the
+                image. Only specified if fig and ax are None. ncols must be also be customized in the same call.
+            ncols(int, optional): designates the number of columns of the figure to plot the different dimensions of the
+                image. Only specified if fig and ax are None. nrows must be also be customized in the same call.
 
-    def plot(self, ax=None, **kwargs):
+        Returns:
+            fig (figure object): figure object in which the graphs are plotted in case ax is None.
+            ax (axes object): axes in which the graphs are plotted.
+
+        """
+        if self.ndim_domain > 2:
+            raise NotImplementedError("Plot only supported for functional data"
+                                      "modeled in at most 3 dimensions.")
+
+        if fig is not None and ax is not None:
+            raise ValueError("fig and axes parameters cannot be passed as arguments at the same time.")
+
+        if fig is not None and len(fig.get_axes()) != self.ndim_image:
+            raise ValueError("Number of axes of the figure must be equal to"
+                             "the dimension of the image.")
+
+        if ax is not None and len(ax)!= self.ndim_image:
+            raise ValueError("Number of axes must be equal to the dimension of the image.")
+
+        if ax is not None and fig is not None and (nrows is not None or ncols is not None):
+            raise ValueError("The number of columns and/or number of rows of the figure, "
+                             "in which each dimension of the image is plotted, can only "
+                             "be customized in case fig is None and ax is None.")
+
+        if (nrows is not None and ncols is None) or (nrows is None and ncols is not None):
```

Comment:
Why? We can use for the other value the minimum that works:
`ncols = math.ceil(self.ndim_image / nrows)`